src/lightning_app/core/app.py

```diff
@@ -596,3 +600,62 @@ def _on_run_end(self):
         if os.getenv("LIGHTNING_DEBUG") == "2":
             del os.environ["LIGHTNING_DEBUG"]
             _console.setLevel(logging.INFO)
+
+    @staticmethod
+    def _extract_vars_from_component_name(component_name: str, state):
+        child = state
+        for child_name in component_name.split(".")[1:]:
+            if child_name in child["flows"]:
+                child = child["flows"][child_name]
+            elif "structures" in child and child_name in child["structures"]:
+                child = child["structures"][child_name]
+            elif child_name in child["works"]:
+                child = child["works"][child_name]
+            else:
+                return None
+
+        return child["vars"]
+
+    def _send_flow_to_work_deltas(self, state) -> None:
+        from lightning_app import LightningWork
+
+        if not self.flow_to_work_delta_queues:
+            return
+
+        for w in self.works:
+            if not w.has_started:
+                continue
+
+            state_work = self._extract_vars_from_component_name(w.name, state)
+            last_state_work = self._extract_vars_from_component_name(w.name, self._last_state)
+
+            # Note: The work was dynamically created or deleted.
+            if state_work is None or last_state_work is None:
+                continue
+
+            # Note: The flow shouldn't update path or drive manually.
+            last_state_work = apply_to_collection(last_state_work, (Path, Drive), lambda x: None)
+            state_work = apply_to_collection(state_work, (Path, Drive), lambda x: None)
+
+            deep_diff = DeepDiff(last_state_work, state_work, verbose_level=2).to_dict()
+
+            if deep_diff:
+                # TODO: Add support for more changes format from DeepDiff.
+                # Currently supported are `values_changed` or `type_changed`.
```

Comment:
This means we are not supporting collection operations correct? (“item_added” / “item_removed” etc)
Let’s raise a proper exception instead of ignoring the change.

Reply:
This would likely break the app for no reason. The feature wasn't support in the past and it shouldn't break. 
But I believe a warning would be acceptable.